var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -1008,7 +1027,10 @@ def cmake_args(self):
         ]
 
         if "spla" in spec and (spec.satisfies("+cuda") or spec.satisfies("+rocm")):
-            args += ["-DCP2K_USE_SPLA_GEMM_OFFLOADING=ON"]
+            args += [
+                "-DCP2K_USE_SPLA_GEMM_OFFLOADING=ON",
+                self.define_from_variant("CP2K_USE_SPLA", "spla"),
```

Comment:
Is it correct to enable SPLA only with `+cuda` or `+rocm`? SLPA also support no GPU backend. Could you please clarify this change?

Reply:
there is no point of using SPLA without GPU backend that's why I only turn it on when GPU support is on.

Reply:
I think it might still be useful for debugging purposes, to reduce the number of libraries using the GPU? If it is supported, I think it would be better to have it as a possibility in Spack too.

Reply:
If you are dead set on this, then then constraint needs to got the the SPLA `variant`/`depends_on` too.